src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1530,7 +1528,12 @@ void getOperandSlots(BoundExpression operand, ArrayBuilder<int> slotBuilder)
                                 // If we got a slot we must have processed the previous conditional receiver.
                                 Debug.Assert(_lastConditionalAccessSlot == -1);
 
-                                slotBuilder.Add(slot);
+                                // We need to continue the walk regardless of whether the receiver is a value
+                                // type, but we only want to update the slots of reference types
+                                if (conditional.Receiver.Type?.IsReferenceType == true)
```

Comment:
Are we handling unconstrained type parameter types?

Reply:
I'll verify this when I address [https://github.com/dotnet/roslyn/issues/28792](https://github.com/dotnet/roslyn/issues/28792), as I'm not changing our existing behavior much here.

---
In reply to: [204568700](https://github.com/dotnet/roslyn/pull/28722#discussion_r204568700) [](ancestors = 204568700)